### PR TITLE
[DOC-11431][DOC-10454] Document admission control for snapshot ingestion

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-11431] Document admission control for snapshot ingestion
[DOC-10454] Docs for ac: disk bandwidth accounting

### CODE REVIEW

File: src/current/v24.3/admission-control.md [hunk 1]
```diff
@@ -44,15 +44,16 @@ Almost all database operations that use CPU or perform storage IO are controlled
...
 - [Bulk data imports]({% link {{ page.version.version }}/import-into.md %}).
-- [Backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}).
-- [Schema changes]({% link {{ page.version.version }}/online-schema-changes.md %}), including index and column backfills (on both the [leaseholder replica]({% link {{ page.version.version }}/architecture/replication-layer.md %}#leases) and [follower replicas]({% link {{ page.version.version }}/architecture/replication-layer.md %}#raft)).
 - [`COPY`]({% link {{ page.version.version }}/copy-from.md %}) statements.
 - [Deletes]({% link {{ page.version.version }}/delete-data.md %}) (including deletes initiated by [row-level TTL jobs]({% link {{ page.version.version }}/row-level-ttl.md %}); the [selection queries]({% link {{ page.version.version }}/selection-queries.md %}) performed by TTL jobs are also subject to CPU admission control).
+- [Backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}).
```
Comment (mdlinville) [inline src/current/v24.3/admission-control.md:49]: Moved these around to make the list make more sense to me -- things that users do regularly, then things that are more complicated / low-level. WDYT?

Comment (rmloveland) [reply src/current/v24.3/admission-control.md:49]: Makes sense to me!

File: src/current/v24.3/admission-control.md [hunk 2]
```diff
@@ -44,15 +44,16 @@ Almost all database operations that use CPU or perform storage IO are controlled
...
 - [Follower replication work]({% link {{ page.version.version }}/architecture/replication-layer.md %}#raft).
 - [Raft log entries being written to disk]({% link {{ page.version.version }}/architecture/replication-layer.md %}#raft).
 - [Changefeeds]({% link {{ page.version.version }}/create-and-configure-changefeeds.md %}).
 - [Intent resolution]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#write-intents).
-
+- {% include_cached new-in.html version="v24.3" %} [Snapshot transfers]({% link {{ page.version.version }}/architecture/replication-layer.md %}#snapshots) onto a node with a [provisioned rate]({% link {{ page.version.version }}/cockroach-start.md %}#store) configured for its store, based on disk bandwidth, to reduce the impact on foreground workloads on the node. Admission control for snapshot transfers is disabled by default. To learn more, refer to [Snapshots]({% link {{ page.version.version }}/architecture/replication-layer.md %}#snapshots).
```
Comment (aadityasondhi) [inline src/current/v24.3/admission-control.md:55]: I think we should either leave this out completely or state that it is an experimental feature in 24.3. cc @dshjoshi .

Comment (mdlinville) [reply src/current/v24.3/admission-control.md]: Removed the line

**Comment by dshjoshi** [comment]
@rmloveland 
4th bullet in this section https://deploy-preview-19068--cockroachdb-docs.netlify.app/docs/v24.3/admission-control.html#enable-and-disable-admission-control   Should say "Preview in 24.3"

Also add 5th bullet  in the same section
"Preview in 24.3" kvadmission.store.provisioned_bandwidth, if set to a non-zero value, this is used as the provisioned bandwidth (in bytes/s), to  enable disk bandwidth based admission control, since admission.disk_bandwidth_tokens.elastic.enabled defaults to true.

**Comment by mdlinville** [comment]
> @rmloveland
> 
> * 4th bullet in this section https://deploy-preview-19068--cockroachdb-docs.netlify.app/docs/v24.3/admission-control.html#enable-and-disable-admission-control   Should say "Preview in 24.3"
> * Also add 5th bullet  in the same section
>   "Preview in 24.3" kvadmission.store.provisioned_bandwidth, if set to a non-zero value, this is used as the provisioned bandwidth (in bytes/s), to  enable disk bandwidth based admission control, since admission.disk_bandwidth_tokens.elastic.enabled defaults to true.
> * Also there is font size formatting issue with the last bullet (intent resolution) in this section https://deploy-preview-19068--cockroachdb-docs.netlify.app/docs/v24.3/admission-control.html#operations-subject-to-admission-control

Done. The missed setting is not marked as New in v24.3.

File: src/current/v24.3/admission-control.md [hunk 3]
```diff
@@ -68,6 +68,8 @@ Admission control is enabled by default. To enable or disable admission control,
 - `admission.kv.enabled` for work performed by the [KV layer]({% link {{ page.version.version }}/architecture/distribution-layer.md %}).
 - `admission.sql_kv_response.enabled` for work performed in the SQL layer when receiving [KV responses]({% link {{ page.version.version }}/architecture/distribution-layer.md %}).
 - `admission.sql_sql_response.enabled` for work performed in the SQL layer when receiving [DistSQL responses]({% link {{ page.version.version }}/architecture/sql-layer.md %}#distsql).
+- {% include_cached new-in.html version="v24.3" %} `kvadmission.store.snapshot_ingest_bandwidth_control.enabled` to optionally limit the disk impact of ingesting snapshots on a node. This cluster setting is in [Preview]({% link {{ page.version.version }}/cockroachdb-feature-availability.md %}#features-in-preview).
+- `kvadmission.store.provisioned_bandwidth` to optionally limit the bandwidth for a store, expressed in bytes per second. This cluster setting is in [Preview]({% link {{ page.version.version }}/cockroachdb-feature-availability.md %}#features-in-preview).
```
Comment (mdlinville) [inline src/current/v24.3/admission-control.md:72]:
```suggestion
- {% include_cached new-in.html version="v24.3" %}`kvadmission.store.provisioned_bandwidth` to optionally limit the disk bandwidth capacity of stores on the cluster. Disk bandwidth admission control paces background disk writes to keep disk bandwidth within its provisioned bandwidth. This cluster setting is in [Preview]({% link {{ page.version.version }}/cockroachdb-feature-availability.md %}#features-in-preview).
```

File: src/current/v24.3/cockroachdb-feature-availability.md [hunk 4]
```diff
@@ -47,6 +47,14 @@ Any feature made available in a phase prior to GA is provided without any warran
 **The following features are in preview** and are subject to change. To share feedback and/or issues, contact [Support](https://support.cockroachlabs.com/hc).
 {{site.data.alerts.end}}
 
+### Admission control for ingesting snapshots
+
+The [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `kvadmission.store.snapshot_ingest_bandwidth_control.enabled`, which allows you to to optionally limit the disk impact of ingesting snapshots on a node, is in Preview.
+
+### Admission control to limit the bandwidth for a store
+
+The [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}), `kvadmission.store.provisioned_bandwidth`, which allows you to optionally limit the bandwidth for a store, expressed in bytes per second, is in [Preview]({% link {{ page.version.version }}/cockroachdb-feature-availability.md %}#features-in-preview).
```
Comment (mdlinville) [inline src/current/v24.3/cockroachdb-feature-availability.md:56]:
```suggestion
The [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}), `kvadmission.store.provisioned_bandwidth`, which allows you to optionally limit the disk bandwidth capacity of stores on the cluster. Disk bandwidth admission control paces background disk writes to keep disk bandwidth within its provisioned bandwidth, is in Preview.
```

**Review by aadityasondhi — APPROVED** [review]

**Comment by mdlinville** [comment]
> > @rmloveland
> > 
> > * 4th bullet in this section https://deploy-preview-19068--cockroachdb-docs.netlify.app/docs/v24.3/admission-control.html#enable-and-disable-admission-control   Should say "Preview in 24.3"
> > * Also add 5th bullet  in the same section
> >   "Preview in 24.3" kvadmission.store.provisioned_bandwidth, if set to a non-zero value, this is used as the provisioned bandwidth (in bytes/s), to  enable disk bandwidth based admission control, since admission.disk_bandwidth_tokens.elastic.enabled defaults to true.
> > * Also there is font size formatting issue with the last bullet (intent resolution) in this section https://deploy-preview-19068--cockroachdb-docs.netlify.app/docs/v24.3/admission-control.html#operations-subject-to-admission-control
> 
> Done. The missed setting is not marked as New in v24.3.

@dshjoshi Is this looking good with the latest changes?

File: src/current/v24.3/architecture/replication-layer.md [hunk 5]
```diff
@@ -116,6 +116,8 @@ Sending data locally using delegated snapshots has the following benefits:
 
 Delegated snapshots are managed automatically by the cluster with no need for user involvement.
 
+{% include_cached new-in.html version="v24.3" %}To limit the impact of snapshot ingestion on a node with a [provisioned rate]({% link {{ page.version.version }}/cockroach-start.md %}#store) configured for its store, you can enable [admission control]({% link {{ page.version.version }}/admission-control.md %}) for snapshot transfer, based on disk bandwidth. This allows you to limit the disk impact on foreground workloads on the node. Admission control for snapshot transfers is disabled by default; to enable it, set the [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `kvadmission.store.snapshot_ingest_bandwidth_control.enabled` to `true`. The historgram [metric]({% link {{ page.version.version }}/metrics.md %}) `admission.wait_durations.snapshot_ingest` allows you to observe the wait times for snapshots that were impacted by admission control.
```
Comment (rmloveland) [inline src/current/v24.3/architecture/replication-layer.md:119]: typo: `s/historgram/histogram/`

File: src/current/v24.3/cockroachdb-feature-availability.md [hunk 6]
```diff
@@ -47,6 +47,14 @@ Any feature made available in a phase prior to GA is provided without any warran
 **The following features are in preview** and are subject to change. To share feedback and/or issues, contact [Support](https://support.cockroachlabs.com/hc).
 {{site.data.alerts.end}}
 
+### Admission control for ingesting snapshots
+
+The [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `kvadmission.store.snapshot_ingest_bandwidth_control.enabled`, which allows you to to optionally limit the disk impact of ingesting snapshots on a node, is in Preview.
+
+### Admission control to limit the bandwidth for a store
+
+The [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}), `kvadmission.store.provisioned_bandwidth`, which allows you to optionally limit the disk bandwidth capacity of stores on the cluster. Disk bandwidth admission control paces background disk writes to keep disk bandwidth within its provisioned bandwidth, is in Preview.
```
Comment (rmloveland) [inline src/current/v24.3/cockroachdb-feature-availability.md:56]: first sentence is missing the "is in preview" clause e.g. "the foo, which allows you to bar, is in preview"

Comment (rmloveland) [reply src/current/v24.3/cockroachdb-feature-availability.md:56]: oh i see the "is in Preview" is at the end of the second sentence.  i think it just needs to go on the end of the first sentence.

**Review by rmloveland — APPROVED** [review]
LGTM thank you Matt for taking this on!!!!

found a couple small non-blocking things, a typo and a missing sentence clause

File: src/current/v24.3/architecture/replication-layer.md (hunk 5 above)
Comment (mdlinville) [reply src/current/v24.3/architecture/replication-layer.md]: Done

File: src/current/v24.3/cockroachdb-feature-availability.md (hunk 6 above)
Comment (mdlinville) [reply src/current/v24.3/cockroachdb-feature-availability.md]: Done

**Pull request merged**